ote_sdk/ote_sdk/test_suite/fixtures.py

```diff
@@ -0,0 +1,361 @@
+
+
+def ROOT_PATH_KEY():  # pylint: disable=invalid-name
+    """
+    Constant for storing in dict-s with paths the root path
+    that will be used for resolving relative paths.
+    """
+    return "_root_path"
+
+
+@pytest.fixture
+def dataset_definitions_fx(request):
+    """
+    Return dataset definitions read from a YAML file passed as the parameter --dataset-definitions.
+
+    Note that the dataset definitions should store the following structure:
+    {
+        <dataset_name1>: { ...<some elements describing dataset1>... },
+        <dataset_name2>: { ...<some elements describing dataset2>... },
+        ...
+    }
+    The elements describing datasets could have arbitrary structure, the
+    structure is defined by the functions parsing dataset in the algo backends.
+
+    An example for mmdetection algo backend:
+    {
+        <dataset_name>: {
+            'annotations_train': <annotation_file_path1>
+            'images_train_dir': <images_folder_path1>
+            'annotations_val': <annotation_file_path2>
+            'images_val_dir': <images_folder_path2>
+            'annotations_test': <annotation_file_path3>
+            'images_test_dir':  <images_folder_path3>
+        }
+    }
+
+    Also one more key with value ROOT_PATH_KEY() is added -- it is the path to
+    the folder where the dataset definitions file is placed, this path will be
+    used to resolve relative paths in the dataset structures.
+    """
+    path = request.config.getoption("--dataset-definitions")
+    if path is None:
+        logger.warning(
+            f"The command line parameter '--dataset-definitions' is not set"
+            f"whereas it is required for the test {request.node.originalname or request.node.name}"
+            f" -- ALL THE TESTS THAT REQUIRE THIS PARAMETER ARE SKIPPED"
+        )
+        return None
+    with open(path, encoding="utf-8") as f:
+        data = yaml.safe_load(f)
+    data[ROOT_PATH_KEY()] = osp.dirname(path)
+    return data
+
+
+def _get_template_name_from_template_file(template_path):
+    with open(template_path, encoding="utf-8") as f:
+        template_data = yaml.safe_load(f)
+        assert "model_template_id" in template_data, f"Wrong template {template_path}"
+        return template_data["model_template_id"]
+
+
+@pytest.fixture(scope="session")
+def template_paths_fx(ote_templates_root_dir_fx):
+    """
+    Return mapping model names to template paths, received from globbing the
+    folder pointed by the fixture ote_templates_root_dir_fx.
+    Note that the function searches files with name `template.yaml`, and for each such file
+    the model name is the name of the parent folder of the file.
+    """
+    root = ote_templates_root_dir_fx
+    assert osp.isabs(
+        root
+    ), f"Error: ote_templates_root_dir_fx is not an absolute path: {root}"
+    glb = glob.glob(f"{root}/**/template*.yaml", recursive=True)
+    data = {}
+    for cur_path in glb:
```

Comment:
please use something meaningful instead of `glb`

Reply:
Fixed